upload-to-ecr/action.yml

```diff
@@ -120,26 +133,129 @@ runs:
         push: true
         cache-from: type=local,src=/tmp/buildx-cache
-    - name: Build final image without additional tag
-      if: steps.login-ecr.outcome == 'success' && !inputs.additional-image-tag
+    # Additional and SHA Tag
+    - name: Build final image with additional tag and SHA tag
+      if: steps.login-ecr.outcome == 'success' && inputs.additional-image-tag && inputs.sha-tag == 1 && !inputs.latest-tag
       uses: docker/build-push-action@v2
       env:
+        ADDITIONAL_IMAGE_TAG: ${{ inputs.additional-image-tag }}
         ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
         ECR_REPOSITORY: ${{ inputs.ecr-repository }}
         IMAGE_TAG: ${{ github.sha }}
+        UUID_TAG: ${{ steps.generate-uuid.outputs.uuid }}
       with:
         context: .
         file: ${{ inputs.dockerfile }}
         builder: ${{ steps.buildx.outputs.name }}
         build-args: ${{ inputs.docker-build-args }}
         tags: |
           ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.IMAGE_TAG }},
+          ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.UUID_TAG }},
+          ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.ADDITIONAL_IMAGE_TAG }}
+        push: true
+        cache-from: type=local,src=/tmp/buildx-cache
+    # Latest Tag
+    - name: Build final image with latest tag only
+      if: steps.login-ecr.outcome == 'success' && !inputs.additional-image-tag && !inputs.sha-tag && inputs.latest-tag == 1
+      uses: docker/build-push-action@v2
+      env:
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        ECR_REPOSITORY: ${{ inputs.ecr-repository }}
+        IMAGE_TAG: ${{ github.sha }}
+        UUID_TAG: ${{ steps.generate-uuid.outputs.uuid }}
+      with:
+        context: .
+        file: ${{ inputs.dockerfile }}
+        builder: ${{ steps.buildx.outputs.name }}
+        build-args: ${{ inputs.docker-build-args }}
+        tags: |
+          ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.UUID_TAG }},
           ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:latest
         push: true
         cache-from: type=local,src=/tmp/buildx-cache
+      # Additional Tag
+    - name: Build final image with additional tag only
+      if: steps.login-ecr.outcome == 'success' && inputs.additional-image-tag && !inputs.sha-tag && !inputs.latest-tag
+      uses: docker/build-push-action@v2
+      env:
+        ADDITIONAL_IMAGE_TAG: ${{ inputs.additional-image-tag }}
+        ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+        ECR_REPOSITORY: ${{ inputs.ecr-repository }}
+        IMAGE_TAG: ${{ github.sha }}
+        UUID_TAG: ${{ steps.generate-uuid.outputs.uuid }}
+      with:
+        context: .
+        file: ${{ inputs.dockerfile }}
+        builder: ${{ steps.buildx.outputs.name }}
+        build-args: ${{ inputs.docker-build-args }}
+        tags: |
+          ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.UUID_TAG }},
```

Comment:
Would it be possible to have a step that outputs a comma-separated list of registry/repo:tag for each tag option passed in, and then uses the output of that in a single `build-push-action`?

I am not sure if I already thought of this idea and it didn't work and that's why I didn't do it, or I just didn't think of it hah.